### PR TITLE
fixes for CIS checks

### LINKED ISSUE #681: CIS Security Benchmark
**Is your feature request related to a problem? Please describe.**
Find a way to easily run the CIS benchmark and document the results. CIS benchmarks are planned to be run for every major and minor release.

**Describe the solution you'd like**
open

**Describe alternatives you've considered**
open

**Additional context**
-

### PR DESCRIPTION
**Pull request by jasmingacic**
This PR is just for findings and discussion only. Based on the findings we can make decisions how to remedy CIS failures/

Fixes #681 

```
1.2.10 Follow the Kubernetes documentation and set the desired limits in a configuration file.
Then, edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
and set the below parameters.
--enable-admission-plugins=...,EventRateLimit,...
--admission-control-config-file=<path/to/configuration/file>

1.2.12 Edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
on the master node and set the --enable-admission-plugins parameter to include
AlwaysPullImages.
--enable-admission-plugins=...,AlwaysPullImages,...

1.2.13 Edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
on the master node and set the --enable-admission-plugins parameter to include
SecurityContextDeny, unless PodSecurityPolicy is already in place.
--enable-admission-plugins=...,SecurityContextDeny,...

1.2.16 Follow the documentation and create Pod Security Policy objects as per your environment.
Then, edit the API server pod specification file /etc/kubernetes/manifests/kube-apiserver.yaml
on the master node and set the --enable-admission-plugins parameter to a
value that includes PodSecurityPolicy:
--enable-admission-plugins=...,PodSecurityPolicy,...
Then restart the API Server.
```

### CODE REVIEW

**Comment by jasmingacic** [comment]
Adding fix for 1.2.10 `--enable-admission-plugins=..., EventRateLimit,...` make k0s cluster fail to start.

**Comment by jasmingacic** [comment]
Added `anonymous-auth=false` to API server parameters

**Comment by jasmingacic** [comment]
Added `terminated-pod-gc-threshold` set to default value 12500 based on these instructions https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

**Comment by jasmingacic** [comment]
Added `	"tls-cipher-suites":                  "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
` to apiserver parameters to comply with 1.2.35

**Comment by jasmingacic** [comment]
I've opened a [discussion](https://github.com/aquasecurity/kube-bench/discussions/847) about adding k0s specific config. Still no response.

**Comment by jasmingacic** [comment]
@jnummelin 
In the PR i've added kube-bench directory in the project root there we have a readme on how to run it and the k0s specific configuration

**Comment by jasmingacic** [comment]
@jnummelin should we leave `kube-check` directory in the project root or we move it elsewhere

**Comment by jnummelin** [comment]
> should we leave kube-check directory in the project root or we move it elsewhere

`kube-bench` has all the needed configs and as such acts as good reference, thus I'd keep it in. But also a short summary on some of the docs pages would make sense.

**Review by mviitane — DISMISSED** [review]
LGTM

**Comment by jasmingacic** [comment]
@jnummelin can you take another look at .go files we changed before we merge this

File: docs/cis_benchmark.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+```
+$ kube-bench run --config-dir docs/kube-bench/cfg/ --benchmark k0s-1.0
+```
+
+## Summary of disabled checks 
+
+### Master Node Security Configuration
+Current configuration has in total of 8 master checks disabled: 
+
+1. **id: 1.2.10**
```
Comment (jnummelin) [inline docs/cis_benchmark.md:24]: This one deserves a note why we skip it

**Review by jnummelin — APPROVED** [review]
Minor nit, otherwise good stuff

**Review by jnummelin — APPROVED** [review]

**Pull request merged**